src/main/functions/DidJWT.ts

```diff
@@ -221,6 +223,12 @@ export function parseJWT(jwt: string): JWTDecoded {
   if (!payload || !header) {
     throw new Error(SIOPErrors.NO_ISS_DID);
   }
+  // Here we have to fetch the presentation_definition part of the decoded jwt
+  if (PEManager.findValidPresentationDefinition(payload)) {
+    payload['peContext'] = PresentationExchangeContext.PE;
```

Comment:
Hmzz, I don't really like this. This is adding things to an existing and signed JWT payload!

This should be done when the payload is being created by the RP

Reply:
To be honest I believe the OP side should simply parse the payload for existence of the PE definition in the payload and store that in the verfiedJwt. Reason is that we are adding logic to our implementation between the OP and RP based upon our own object. This means the OP could not work anymore with an RP using another library but conforming to SIOP v2